lib/bundler/inject.rb

```diff
@@ -33,6 +33,24 @@ def skip_warnings?
           bundler_setting
         end
       end
+
+      # Location where we can find overriden gems
+      #
+      # This can be set in two ways:
+      #
+      #   - Via bundler's Bundler::Settings
+      #
+      # To configure the setting, you can run:
+      #
+      #   bundle config bundler_inject.gem_path ~/src:~/gems_src
+      #
+      # OR use an environment variable
+      #
+      #   BUNDLE_BUNDLER_INJECT__GEM_PATH=~/src:~/gems_src
+      #
+      def gem_path
+        @gem_path ||= (Bundler.settings["bundler_inject.gem_path"] || "").split(File::PATH_SEPARATOR)
```

Comment:
Should we also `.map { |p| File.expand_path(p) }`, on the end of this to expand any `~`?